pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderToken.java

```diff
@@ -409,4 +444,134 @@ public String getHeader(String name) {
             return super.getHeader(name);
         }
     }
+
+    private static final class TokenSigningKeyResolver implements SigningKeyResolver {
```

Comment:
I think the `TokenSigningKeyResolver` should be called in the `readKeyFromUrl` function